controllers/workspace/devworkspace_controller.go

```diff
@@ -281,9 +281,15 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 	workspace.Spec.Template = *flattenedWorkspace
 
-	err = ssh.AddSshAgentPostStartEvent(&workspace.Spec.Template)
-	if err != nil {
-		return r.failWorkspace(workspace, "Failed to add ssh-agent post start event", metrics.ReasonWorkspaceEngineFailure, reqLogger, &reconcileStatus), nil
+	if needsSSHAgentPostStartEvent, err := ssh.NeedsSSHPostStartEvent(clusterAPI, workspace.Namespace); err != nil {
```

Comment:
@AObuchow @dkwon17 to be on the safe side can we enable this feature based on DWOC property?

Reply:
based on the standup's discussion I like the idea of enabling it based on 

```
apiVersion: controller.devfile.io/v1alpha1
config:
  enableExperimentalFeatures: true
```

Reply:
Unfortunately, the additional commit I created for guarding the SSH agent initialization postStart event injection with `enableExperimentalFeatures` got merged into my original commit while I was cleaning up the git log. However, this change is [present](https://github.com/devfile/devworkspace-operator/pull/1341/files#diff-05e72230161a4d41c5a27fb453041c78839c193a9ddcb5066b1cb62c5a399529R284).